### PR TITLE
[YUNIKORN-1818] Update developer guide about deployment

### PR DESCRIPTION
**Pull request by yzhangal**
### What is this PR for?
A few sentences describing the overall goals of the pull request's commits.
First time? Check out the contributing guide - http://yunikorn.apache.org/community/how_to_contribute   

### What type of PR is it?
* [ ] - Bug Fix
* [x ] - Improvement
* [ ] - Feature
* [x ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/YUNIKORN-1818 Update developer guide about deployment

### How should this be tested?
Run "make test"

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Review by wilfred-s — CHANGES_REQUESTED** [review]
There is a second file that has the same volume mounting: scheduler-load.yaml needs to be updated also.
The README also points to loading the config map that should be removed as all setups do that now.

**Comment by yzhangal** [comment]
> scheduler-load.yaml

Thanks @wilfred-s which README file?

File: test/e2e/framework/helpers/yunikorn/wrappers.go [hunk 1]
```diff
@@ -131,8 +131,7 @@ func RestoreConfigMapWrapper(oldConfigMap *v1.ConfigMap, annotation string) {
 	var e, err3 = k.UpdateConfigMap(c, configmanager.YuniKornTestConfig.YkNamespace)
 	Ω(err3).NotTo(HaveOccurred())
 	Ω(e).NotTo(BeNil())
-	// Updating scheduler pod annotation to trigger force refresh of configmaps
-	Ω(k.RemoveYunikornSchedulerPodAnnotation(annotation)).NotTo(HaveOccurred())
```
Comment (FrankYang0529) [inline test/e2e/framework/helpers/yunikorn/wrappers.go:135]: It looks like `KubeCtl.RemoveYunikornSchedulerPodAnnotation` is only called here. We can also remove it.

https://github.com/apache/yunikorn-k8shim/blob/292e26fd9d910be39bb98f3bc01b7445179dff9e/test/e2e/framework/helpers/k8s/k8s_utils.go#L945

**Review by FrankYang0529 — COMMENTED** [review]
If we don't need to update annotation to trigger pod refresh configuration, we can also remove `UpdateYunikornSchedulerPodAnnotation` in `UpdateCustomConfigMapWrapper`.

https://github.com/apache/yunikorn-k8shim/blob/292e26fd9d910be39bb98f3bc01b7445179dff9e/test/e2e/framework/helpers/yunikorn/wrappers.go#L109-L110

**Comment by yzhangal** [comment]
> > scheduler-load.yaml
> 
> Thanks @wilfred-s which README file?

HI @wilfred-s did you mean the "Configuration loaded from config map" in ./deployments/scheduler/README.md below and suggest to remove "Configuration loaded from config map"? Thanks

Deploys scheduler-core + scheduler-web

* Scheduler pod runs 2 containers, 1 for scheduler-core, 1 for scheduler-web
* Configuration loaded from config map
* UI port is `9889`

## Load Balancer

Deploys scheduler-core + scheduler-web

* Scheduler pod runs 2 containers, 1 for scheduler-core, 1 for scheduler-web
* Configuration loaded from config map
* UI port is `9889`
* A load balancer that helps to expose web UI link directly on K8s

File: test/e2e/framework/helpers/yunikorn/wrappers.go (hunk 1 above)
Comment (yzhangal) [reply test/e2e/framework/helpers/yunikorn/wrappers.go:135]: Thanks @FrankYang0529 I will update in next rev.

**Comment by yzhangal** [comment]
> > > scheduler-load.yaml
> > 
> > 
> > Thanks @wilfred-s which README file?
> 
> HI @wilfred-s did you mean the "Configuration loaded from config map" in ./deployments/scheduler/README.md below and suggest to remove "Configuration loaded from config map"? Thanks
> 
> Deploys scheduler-core + scheduler-web
> 
> * Scheduler pod runs 2 containers, 1 for scheduler-core, 1 for scheduler-web
> * Configuration loaded from config map
> * UI port is `9889`
> 
> ## Load Balancer
> Deploys scheduler-core + scheduler-web
> 
> * Scheduler pod runs 2 containers, 1 for scheduler-core, 1 for scheduler-web
> * Configuration loaded from config map
> * UI port is `9889`
> * A load balancer that helps to expose web UI link directly on K8s

I removed the above two lines in the new PR rev. Thanks.

File: test/e2e/framework/helpers/yunikorn/wrappers.go (hunk 1 above)
Comment (yzhangal) [reply test/e2e/framework/helpers/yunikorn/wrappers.go:135]: Done in the new PR rev submitted, thanks.

File: test/e2e/framework/helpers/yunikorn/wrappers.go [hunk 2]
```diff
@@ -106,8 +106,6 @@ func UpdateCustomConfigMapWrapper(oldConfigMap *v1.ConfigMap, schedPolicy string
 	Ω(err3).NotTo(HaveOccurred())
 	Ω(d).NotTo(BeNil())
 
-	// Updating scheduler pod annotation to trigger force refresh of configmaps
-	Ω(k.UpdateYunikornSchedulerPodAnnotation(annotation)).NotTo(HaveOccurred())
```
Comment (FrankYang0529) [inline test/e2e/framework/helpers/yunikorn/wrappers.go:110]: The `KubeCtl.UpdateYunikornSchedulerPodAnnotation` is only used here. We can also remove it. Thanks.

https://github.com/apache/yunikorn-k8shim/blob/292e26fd9d910be39bb98f3bc01b7445179dff9e/test/e2e/framework/helpers/k8s/k8s_utils.go#L927

Comment (wilfred-s) [reply test/e2e/framework/helpers/yunikorn/wrappers.go:110]: yes the definition of this function should be removed too `UpdateYunikornSchedulerPodAnnotation()`

Comment (yzhangal) [reply test/e2e/framework/helpers/yunikorn/wrappers.go:110]: Thanks, the latest PR rev has this suggested change.

**Review by wilfred-s — APPROVED** [review]
LGTM, thank you @FrankYang0529 for the help reviewing and @yzhangal for the work

**Pull request closed without merging**